wireless_modules/middle_module/MQTT_Client_class.py

```
@@ -0,0 +1,89 @@
+        :param topics_to_subscribe: An array of topics to subscribe to.
+                                    Eah element must be a string or byte literal (the latter is preferred)
+        :param callback_func: The function to be called whenever a message from the subscribed topics is received.
+        :return: True if a connection to the Broker is successfully established, otherwise False
+        """
+
+        self.client.set_callback(callback_func)
+        try:
+            # Connect to MQTT broker
+            self.client.connect()
+            print('Connected to %s MQTT broker' % (self.mqtt_broker))
+        except OSError:
+            return False
+
+        # Subscribe to each topic
+        for topic in topics_to_subscribe:
+            self.client.subscribe(topic)
+            print('Subscribed to %s topic' % (topic))
+
+        return True
+
+    def _to_bytes_literal(self, data):
+        """
+        Converts 'data' into a form MQTT can read
+        :param data: A string of data to convert to bytes literal
+        :return: The bytes literal version of the 'data'
+        """
+        str_data = str(data)
+        return str.encode(str_data)
+
+    def mqtt_pub(self, data, topic):
+        """
+        This functions takes care of all of the formatting and publishes 'data' on the given 'topic'. It also checks for
+        any pending message, but this should not be relied upon solely - use the check_for_message() method externally
+        when an incoming message is to be checked for/read.
+        :param data: A string of data to be sent
+        :param topic: A string representing the topic to send 'data' to.
+        :return: Nothing
+        """
+        try:
+            msg = self._to_bytes_literal(data)
+
+            self.client.publish(topic, msg)
+
+            self.check_for_message()
+
+        except OSError:
+            print('OSError, check line 73 of MQTT_Client_class.py')
```

Again, better to let the error be thrown - that way we won't even have to worry about noting the line number, the traceback will do that for us.